packages/composite-checkout/src/lib/payment-methods/eps.js

```diff
@@ -0,0 +1,263 @@
+		},
+		getCustomerBank( state ) {
+			return state.customerBank || '';
+		},
+	};
+
+	const store = registerStore( 'eps', {
+		reducer(
+			state = {
+				customerName: { value: '', isTouched: false },
+				customerBank: { value: '', isTouched: false },
+			},
+			action
+		) {
+			switch ( action.type ) {
+				case 'CUSTOMER_NAME_SET':
+					return { ...state, customerName: { value: action.payload, isTouched: true } };
+				case 'CUSTOMER_BANK_SET':
+					return { ...state, customerBank: { value: action.payload, isTouched: true } };
+			}
+			return state;
+		},
+		actions,
+		selectors,
+	} );
+
+	return { ...store, actions, selectors };
+}
+
+export function createEpsMethod( { store, stripe, stripeConfiguration } ) {
+	return {
+		id: 'eps',
+		label: <EpsLabel />,
+		activeContent: <EpsFields stripe={ stripe } stripeConfiguration={ stripeConfiguration } />,
+		submitButton: (
+			<EpsPayButton store={ store } stripe={ stripe } stripeConfiguration={ stripeConfiguration } />
+		),
+		inactiveContent: <EpsSummary />,
+		getAriaLabel: ( __ ) => __( 'EPS e-Pay' ),
+	};
+}
+
+function EpsFields() {
+	const { __ } = useI18n();
+
+	const customerName = useSelect( ( select ) => select( 'eps' ).getCustomerName() );
+	const { changeCustomerName } = useDispatch( 'eps' );
+	const { formStatus } = useFormStatus();
+	const isDisabled = formStatus !== 'ready';
+
+	return (
+		<EpsFormWrapper>
+			<EpsField
+				id="cardholderName"
+				type="Text"
+				autoComplete="cc-name"
+				label={ __( 'Your name' ) }
+				value={ customerName?.value ?? '' }
+				onChange={ changeCustomerName }
+				isError={ customerName?.isTouched && customerName?.value.length === 0 }
+				errorMessage={ __( 'This field is required' ) }
+				disabled={ isDisabled }
+			/>
+		</EpsFormWrapper>
+	);
+}
+
+const EpsFormWrapper = styled.div`
+	padding: 16px;
+	position: relative;
+
+	:after {
+		display: block;
+		width: calc( 100% - 6px );
+		height: 1px;
+		content: '';
+		background: ${ ( props ) => props.theme.colors.borderColorLight };
+		position: absolute;
+		top: 0;
+		left: 3px;
+
+		.rtl & {
+			left: auto;
+			right: 3px;
+		}
+	}
+`;
+
+const EpsField = styled( Field )`
+	margin-top: 16px;
+
+	:first-of-type {
+		margin-top: 0;
+	}
+`;
+
+function EpsPayButton( { disabled, store, stripe, stripeConfiguration } ) {
+	const { __ } = useI18n();
+	const [ items, total ] = useLineItems();
+	const { formStatus } = useFormStatus();
+	const {
+		setTransactionRedirecting,
+		setTransactionError,
+		setTransactionPending,
+	} = useTransactionStatus();
+	const submitTransaction = usePaymentProcessor( 'eps' );
+	const onEvent = useEvents();
+	const customerName = useSelect( ( select ) => select( 'eps' ).getCustomerName() );
+
+	return (
+		<Button
+			disabled={ disabled }
+			onClick={ () => {
+				if ( isFormValid( store ) ) {
+					debug( 'submitting eps payment' );
+					setTransactionPending();
+					onEvent( { type: 'REDIRECT_TRANSACTION_BEGIN' } );
```

Comment:
This event requires a payload to specify the type, like this.
```suggestion
					onEvent( { type: 'REDIRECT_TRANSACTION_BEGIN', payload: { paymentMethodId: 'eps' } } );
```